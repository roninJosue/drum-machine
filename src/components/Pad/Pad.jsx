import React from 'react';
import PropTypes from 'prop-types';
import styled from "styled-components";
import {borderColor, primaryColor, primaryColorHover, white} from "../../utils/colors";

const StyledButton = styled.button`
  border-radius: .5rem;
  background-color: ${white};
  padding: 2rem;
  color: ${primaryColor};
  text-transform: uppercase;
  border: 1px solid ${borderColor};
  font-size: 1.2rem;
  font-weight: bolder;
  border-bottom: 4px solid ${borderColor};
  transition: all .500ms ease;

  &:hover {
    background-color: ${primaryColor};
    cursor: pointer;
    color: ${white};
    border-color: ${primaryColorHover};
  }
  
  &:active {
    border-bottom: none;
    border-top: 4px solid ${primaryColorHover};
  }
`

const Pad = ({text, onClick}) => {
  return (
    <StyledButton
      onClick={onClick}
      className='drum-pad'
    >
      {text}
    </StyledButton>
  );
};

Pad.propTypes = {
  text: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired
};

export default Pad;
