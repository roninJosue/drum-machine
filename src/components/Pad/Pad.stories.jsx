import React from "react"
import Pad from './Pad'

export default {
  title: 'Pad',
  component: Pad
}

export const PadExample = () => (<Pad onClick={() => {}} text='e' />)