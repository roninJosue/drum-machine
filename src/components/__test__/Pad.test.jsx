import React from "react"
import {render, screen} from "@testing-library/react"
import Pad from "../Pad/Pad";

describe('Pad component', () => {
  it('should render', function () {
    const fn = jest.fn()
    const {asFragment} = render(<Pad onClick={fn} text='A' />)

    const html = asFragment()

    expect(html).toMatchSnapshot()
  });
})