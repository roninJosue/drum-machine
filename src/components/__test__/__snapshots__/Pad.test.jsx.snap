// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pad component should render 1`] = `
<DocumentFragment>
  <button
    class="sc-bdvvtL hnBZed drum-pad"
  >
    A
  </button>
</DocumentFragment>
`;
